tests/unit/dns_test.cc

```diff
@@ -36,6 +36,8 @@ using namespace seastar::net;
 
 static const sstring seastar_name = "seastar.io";
 
+#if SEASTAR_TESTING_WITH_NETWORKING
```

Comment:
@Comrade88 ,  hi Galas, thank you for this fix! You've correctly identified that several tests need proper handling with the `SEASTAR_TESTING_WITH_NETWORKING` macro.

While your approach works, I've taken the initiative to implement an alternative solution in PR #2683 that centralizes all the network-dependency flags in a single location. This should improve code readability and maintenance by avoiding scattered conditional checks throughout the codebase.

Could you please take a look at the alternative implementation when you have a chance. I believe it addresses the same issue you've identified while providing a more consolidated approach to managing network-dependent tests.